app/models/miq_server/server_smart_proxy.rb

```diff
@@ -11,17 +11,13 @@ module ClassMethods
     # Called from VM scan job as well as scan_sync_vm
 
     def use_broker_for_embedded_proxy?(type = nil)
-      cores_settings = MiqServer.my_server.get_config("vmdb").config[:coresident_miqproxy].dup
-      result = ManageIQ::Providers::Vmware::InfraManager.use_vim_broker? && cores_settings[:use_vim_broker]
+      result = ManageIQ::Providers::Vmware::InfraManager.use_vim_broker? &&
+               ::Settings.coresident_miqproxy[:use_vim_broker]
+      return result if type.blank? || !result
 
       # Check for a specific type (host/ems) if passed
-      unless type.blank?
-        # Default use_vim_broker to true for ems type
-        cores_settings[:use_vim_broker_ems] = true if cores_settings[:use_vim_broker_ems].blank? && cores_settings[:use_vim_broker_ems] != false
-        cores_settings["use_vim_broker_#{type}".to_sym] ||= false
```

Comment:
Updating global ::Settings is bad (I found couple  other places, will create separate PR) since update will be there up until next ```::Settings.reload```.  In original code ```core_settings``` was ```.dup``` from global, so it was fine.